File: app/src/main/java/com/nollpointer/dates/api/WikipediaResponseModel.java
package com.nollpointer.dates.api;

import com.google.gson.annotations.Expose;
import com.google.gson.annotations.SerializedName;

public class WikipediaResponseModel {

    @SerializedName("title")
    @Expose
    private String title;
    @SerializedName("displaytitle")
    @Expose
    private String displaytitle;
    @SerializedName("pageid")
    @Expose
    private Integer pageid;
    @SerializedName("extract")
    @Expose
    private String extract;
    @SerializedName("extract_html")
    @Expose
    private String extractHtml;
    @SerializedName("lang")
    @Expose
    private String lang;
    @SerializedName("dir")
    @Expose
    private String dir;
    @SerializedName("timestamp")
    @Expose
    private String timestamp;
    @SerializedName("description")
    @Expose
    private String description;
    @SerializedName("content_urls")
    @Expose
    private ContentUrls contentUrls;

    public String getTitle() {
        return title;
    }

    public void setTitle(String title) {
        this.title = title;
    }

    public String getDisplaytitle() {
        return displaytitle;
    }

    public void setDisplaytitle(String displaytitle) {
        this.displaytitle = displaytitle;
    }

    public Integer getPageid() {
        return pageid;
    }

    public void setPageid(Integer pageid) {
        this.pageid = pageid;
    }

    public String getExtract() {
        return extract;
    }

    public void setExtract(String extract) {
        this.extract = extract;
    }

    public String getExtractHtml() {
        return extractHtml;
    }

    public void setExtractHtml(String extractHtml) {
        this.extractHtml = extractHtml;
    }

    public String getLang() {
        return lang;
    }

    public void setLang(String lang) {
        this.lang = lang;
    }

    public String getDir() {
        return dir;
    }

    public void setDir(String dir) {
        this.dir = dir;
    }

    public String getTimestamp() {
        return timestamp;
    }

    public void setTimestamp(String timestamp) {
        this.timestamp = timestamp;
    }

    public String getDescription() {
        return description;
    }

    public void setDescription(String description) {
        this.description = description;
    }

    public ContentUrls getContentUrls() {
        return contentUrls;
    }

    public void setContentUrls(ContentUrls contentUrls) {
        this.contentUrls = contentUrls;
    }

}

File: app/src/main/java/com/nollpointer/dates/api/WikipediaResponseModel.kt
package com.nollpointer.dates.api

import com.google.gson.annotations.Expose
import com.google.gson.annotations.SerializedName

class WikipediaResponseModel {
    @SerializedName("title")
    @Expose
    var title: String? = null
    @SerializedName("displaytitle")
    @Expose
    var displaytitle: String? = null
    @SerializedName("pageid")
    @Expose
    var pageid: Int? = null
    @SerializedName("extract")
    @Expose
    var extract: String? = null
    @SerializedName("extract_html")
    @Expose
    var extractHtml: String? = null
    @SerializedName("lang")
    @Expose
    var lang: String? = null
    @SerializedName("dir")
    @Expose
    var dir: String? = null
    @SerializedName("timestamp")
    @Expose
    var timestamp: String? = null
    @SerializedName("description")
    @Expose
    var description: String? = null
    @SerializedName("content_urls")
    @Expose
    var contentUrls: ContentUrls? = null

}